metrics/internal/collectors/object-bucket.go

```diff
@@ -0,0 +1,226 @@
+func (c *OBCollector) Collect(ch chan<- prometheus.Metric) {
+	cephObjectStoreLister := cephv1listers.NewCephObjectStoreLister(c.Informer.GetIndexer())
+	cephObjectStores := getAllObjectStores(cephObjectStoreLister, c.AllowedNamespaces)
+	if len(cephObjectStores) > 0 {
+		for _, cephObjectStore := range cephObjectStores {
+			//TODO: SSL endpoint
+			adminAPI := c.GetAdminOpsClient(*cephObjectStore)
+			if (cephObjectStore.Spec.Gateway.Port > 0) && adminAPI != nil {
+				c.collectObjectBucketMetrics(*cephObjectStore, adminAPI, ch)
+			}
+		}
+	}
+}
+
+func getEndpoint(cephObjectStore cephv1.CephObjectStore) (host string, port int) {
+	return fmt.Sprintf("%s-%s.%s.%s", rgwServiceName, cephObjectStore.Name, cephObjectStore.Namespace, svcDNSSuffix), int(cephObjectStore.Spec.Gateway.Port)
+}
+func (c *OBCollector) getAllObjectBuckets(cephObjectStore cephv1.CephObjectStore) (objectBuckets []libbucket.ObjectBucket) {
+	selector := fmt.Sprintf("bucket-provisioner=%s.%s", cephObjectStore.Namespace, bucketProvisionerName)
+	listOpts := metav1.ListOptions{
+		LabelSelector: selector,
+	}
+
+	objectBucketsall, err := c.bktclient.ObjectbucketV1alpha1().ObjectBuckets().List(context.TODO(), listOpts)
+	if err != nil {
+		klog.Errorf("couldn't list ObjectBuckets. %v", err)
+		return
+	}
+	objectBuckets = make([]libbucket.ObjectBucket, 0)
+
+	//filter based on ceph-object store
+	bucketHost, _ := getEndpoint(cephObjectStore)
+	for _, ob := range objectBucketsall.Items {
+		if ob.Spec.Endpoint.BucketHost == bucketHost {
+			objectBuckets = append(objectBuckets, ob)
+		}
+	}
+	return
+}
+
+func (c *OBCollector) GetAdminOpsClient(cephObjectStore cephv1.CephObjectStore) *rgwadmin.API {
+	ctx := context.TODO()
+	prometheusSecretName := fmt.Sprintf("rook-ceph-object-user-%s-%s", cephObjectStore.Name, prometheusUserName)
+	secret, _ := c.k8sclient.CoreV1().Secrets(cephObjectStore.Namespace).Get(ctx, prometheusSecretName, metav1.GetOptions{})
+
+	if secret != nil {
+		host, port := getEndpoint(cephObjectStore)
+		endPoint := fmt.Sprintf("http://%s:%d", host, port)
+		adminAPI, err := rgwadmin.New(endPoint, string(secret.Data[accessKey]), string(secret.Data[secretKey]), nil)
+		if err != nil {
+			klog.Errorf("rgwadminapi connection failure error %v", err)
+		}
+		return adminAPI
+	}
+	/* if secret is not found, do we need to create user from here or let OCS-Op automate it?
+		    objectUser := rookclient.CephObjectStoreUser{
+	        ObjectMeta: metav1.ObjectMeta{
+	            Name:      prometheusUserName,
+	            Namespace: cephObjectStore.Namespace,
+	        },
+	        Spec: cephv1.ObjectStoreUserSpec{
+	            Store: cephObjectStore.Name,
+	        },
+	        TypeMeta: metav1.TypeMeta{
+	            Kind: "CephObjectStoreUser",
+	        },
+	    }
+
+			prometheusUser, err := c.rookclient.CephV1().CephObjectStoreUsers(cephObjectStore.Namespace).Create(ctx, objectUser, metav1.CreateOptions{})
+	*/
```

Comment:
remove this if not needed.

Reply:
@umangachapagain : do we create the user if the secret is not present, otherwise it will add another step in documentation to create this user. Same reason I didn't handle the error atm. Please share your opinon